src/Tasks/AssemblyDependency/Resolver.cs

```diff
@@ -181,7 +196,27 @@ ResolutionSearchLocation searchLocation
 
             bool isSimpleAssemblyName = assemblyName?.IsSimpleName == true;
 
-            if (fileExists(pathToCandidateAssembly))
+            bool fileFound;
+            if (useDirectoryCache && Utilities.ChangeWaves.AreFeaturesEnabled(Utilities.ChangeWaves.Wave16_10))
+            {
+                // this verifies file existence using fileExistsInDirectory delegate which internally used cached set of all files in a particular directory
+                // if some cases it render better performance than one by one FileExists
+                try
+                {
+                    fileFound = fileExistsInDirectory(directory, fileName);
+                }
+                catch (Exception e) when (ExceptionHandling.IsIoRelatedException(e))
+                {
+                    // Assuming it's the search path that's bad. But combine them both so the error is visible if it's the reference itself.
```

Comment:
What does it mean for the search path to be bad? That it has invalid characters? That's the only time I could imagine it throwing.

Reply:
yes, it is invalid character, and it throws only on full .Net Framework.